arborcat.module

```diff
@@ -1,120 +1,216 @@
-        if ($matches[1][0]>0) {
-          $available = TRUE;
-        }
+      return $available;
+  }
+
+  /*
+   * returns a list of request pickup locations.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function arborcat_pickupLocations($destLocation = null)
+  {
+      $db = \Drupal::database();
+      $query = $db->select('arborcat_pickup_location', 'apl');
+      $query->fields('apl', ['locationId', 'branchLocationId', 'timePeriod', 'timePeriodStart', 'timePeriodEnd', 'maxLockers', 'locationName', 'locationDescription']);
+        
+      // add in a condition if a location is supplied to filter on
+      if (3 == strlen($destLocation)) {
+          $query->condition('apl.branchLocationId', (int) $destLocation, '=');
       }
-    return $available;
+      $result = $query->execute();
+      
+      $pickupLocationRecords = $result->fetchAll();
+      return $pickupLocationRecords;
+  }
+
+   /*
+   * Checks the availability of a locker fo the specified date and timePeriod.
+   * If a branch location is passed in, that a filtered list of pickup locations is returned
+   */
+  function lockerAvailableForDateAndTimeSlot($queryDate, $locationObject)
+  {
+      $availableLockers = 0;
+      // Query current arborcat_patron_pickup_request records for the date and timePeriod to get inuse count
+      // fields pickupDate & timeSlot
+      $db = \Drupal::database();
+      $timeperiod = $locationObject->timePeriod;
+      $query = $db->select('arborcat_patron_pickup_request', 'appr');
+      $query-> fields('appr', ['id']);
+      $query-> condition('pickupDate', $queryDate);
+      $query-> condition('timeSlot', $timeperiod, '=');
+      $results = $query->execute()->fetchAll();
+
+      $lockersInUse = count($results);
+      $availableLockers = $locationObject->maxLockers - $lockersInUse;
+      
+      return ($availableLockers > 0);
   }
-  
-?>
+
+function addPickupRequest($pickupLocation, $pickupDay, $timeSlot, $contactEmail, $contactPhone, $contactSMS)
+{
+    $result = [];
+    
+    $user = \Drupal\user\Entity\User::load(\Drupal::currentUser()->id());
+    if ($user->isAuthenticated()) {
+        $db = \Drupal::database();
+        $db->insert('arborcat_patron_pickup_request')
+        ->fields([
+            'uid' => $user->id(),
+            'pickupDay' => $pickupDay,
+            'timeSlot' => $timeSlot,
+            'pickupLocation' => $pickupLocation,
+            'contactEmail' => $contactEmail,
+            'contactPhone' => $contactPhone,
+            'contactSMS' => $contactSMS,
+            'timestamp' => time()
+        ])
+        ->execute();
+        $result['success'] = "Successfully added pickup request";
+    } else {
+        $result['error'] = 'You must be logged in to make a pickup request.';
+    }
+    return $result;
+}
+
+/*
+    * Debugging routine to log to the <root folder>/LWKLWK.log
+    */
+    function dblog(...$thingsToLog)
```

Comment:
This looks like a debug function for the programmer, not sure it should be rolled into production. If a log is needed, I'd rename the function to something less generic